src/main/kotlin/app/CodeLongevity.kt

```diff
@@ -0,0 +1,231 @@
+// Copyright 2017 Sourcerer Inc. All Rights Reserved.
+
+package app
+
+import org.eclipse.jgit.diff.DiffFormatter
+import org.eclipse.jgit.diff.DiffEntry
+import org.eclipse.jgit.diff.RawText
+import org.eclipse.jgit.internal.storage.file.FileRepository
+import org.eclipse.jgit.lib.ObjectId
+import org.eclipse.jgit.lib.ObjectLoader
+import org.eclipse.jgit.lib.Repository
+import org.eclipse.jgit.revwalk.RevCommit
+import org.eclipse.jgit.revwalk.RevWalk
+import org.eclipse.jgit.treewalk.TreeWalk
+import org.eclipse.jgit.util.io.DisabledOutputStream
+
+import java.util.Date
+
+/**
+ * Identifies a code line in a file in a revision.
+ */
+class RevCommitLine(val commit: RevCommit, val file: String, val line: Int)
+
+/**
+ * Represents a code line.
+ *
+ * TODO(Alex): the text arg is solely for testing proposes (remove it)
+ */
+class CodeLine(val from: RevCommitLine, val to: RevCommitLine, val text: String) {
+
+    // XXX(Alex): oldId and newId may be computed as a hash built from commit,
+    // file name and line number, if we are going to send the data outside a
+    // local machine.
+
+    /**
+     * Id of the code line in a revision when the line was added. Used to
+     * update the line's lifetime computed in previous iterations.
+     * 
+     */
+    val oldId: String = ""
+
+    /**
+     * Id of the line in head revision (if this is an existing line) or
+     * a revision where the line was deleted.
+     */
+    val newId: String = ""
+
+    /**
+     * A line age in seconds.
+     */
+    val age = to.commit.getCommitTime() - from.commit.getCommitTime()
+
+    /**
+     * A pretty print of a code line; debugging.
+     */
+    fun printme() {
+        val fd = Date(from.commit.getCommitTime().toLong() * 1000).toLocaleString()
+        val fc = "${from.commit.getName()} '${from.commit.getShortMessage()}'"
+        val tc = "${to.commit.getName()} '${to.commit.getShortMessage()}'"
+        val td = Date(to.commit.getCommitTime().toLong() * 1000).toLocaleString()
+        println("Line '$text' - '${from.file}:${from.line}' added in $fc $fd")
+        println("  last known as '${to.file}:${to.line}' in $tc $td")
+    }
+}
+
+/**
+ * Used to compute age of code lines in the repo.
+ */
+class CodeLongevity(repoPath: String, tailRev: String) {
+    val repo = FileRepository(repoPath)
+    val head: RevCommit = RevWalk(repo).parseCommit(repo.resolve("HEAD"));
+    val tail: RevCommit? =
+        if (tailRev != "") RevWalk(repo).parseCommit(repo.resolve(tailRev))
+        else null
+
+    var codeLines: MutableList<CodeLine> = mutableListOf()
+
+    init {
+        compute()
+    }
+
+    fun test(email: String) {
```

Comment:
it's not a testing suit :) check out the pull request description, I left a more generic question about it